### ISSUE TITLE
[webview_flutter] Accessibility element not focused when trying to select the element directly.

### ISSUE THREAD

**Issue by cyanglaz** [flutter/flutter#30804]
When using a11y on iOS webview, we are able to focus on accessibility elements by going through all the elements one by one.

However, if we want to directly select an element inside the webview, by dragging a finger on the webview.

This issue can be reproduced directly using the example app in webview_flutter plugin.

**Comment by cyanglaz** [comment]
https://github.com/flutter/engine/pull/8731 should fix the issue for platform views in general. However, WKWebView is still not working with the fix.

**Comment by cyanglaz** [comment]
Tried with UIWebView and it is working. Seems like there could be some internal bug in WKWebView which fails when adding WKWebView into a custom configured a11y tree. I created a sample pure iOS app to reproduced the issue and sent a bug report to Apple.
There might be some workaround that I haven't found, other than that, this might be blocked until we get anything back from Apple.

**Comment by cyanglaz** [comment]
I have filed a bug with Apple and see if Apple can fix this in the WKWebView. Meanwhile, I have filed a stack overflow question (https://stackoverflow.com/questions/56570808/how-to-correctly-add-wkwebview-to-accessibilityelements-in-an-accessibilityconta). Hopefully we can get some feedbacks from the iOS community. 

I have also come up with a workaround which requires a major change of our current accessibility implementation. I will start work on the workaround if it gets approved by the team.

**Comment by cyanglaz** [comment]
There are some other preparation works need to be done in order to work on this, including: 
1. Making scenario app tests running on physical devices, so that we can test a11y on iOS.
https://github.com/flutter/flutter/issues/43852
2. Adding a device lab iPhone that is dedicated for testing a11y which has `voice over` on .
https://github.com/flutter/flutter/issues/42270
3. Adding accessibility bridge test so we are sure the fix provided here does not regress a11y bridge.
https://github.com/flutter/flutter/issues/42669

I'm updating the current goal as it is not likely all of the above and the actual fix of this issue can be done by December 2019.

**Comment by cyanglaz** [comment]
Confirmed that the issue is gone in iOS 13.2.3 and above. Since the issue now only exists in old iOS versions, I'm going to lower the priority of the issue. I will also un-assign myself from the issue as I am not going to actively working on it in near term.

**Comment by danagbemava-nc** [comment]
I am able to select an element by dragging over the element and also selecting them individually on `iOS 14.6` with `stable 2.2.3`.
 
[RPReplay_Final1629195054 2.mov.zip](https://github.com/flutter/flutter/files/6999145/RPReplay_Final1629195054.2.mov.zip)

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-arm, locale
    en-GH)
    • Flutter version 2.2.3 at /Users/nexus/dev/sdks/flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        •
      android-arm64  • Android 12 (API 31) (emulator)
    • iPhone 12 (mobile)          • F2592D43-5FE3-46BB-8A2B-A38BA468DCE9 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)                • chrome                               •
      web-javascript • Google Chrome 92.0.4515.131

• No issues found!
```
</details>

cc: @cyanglaz should this issue still be kept open for tracking?

**Comment by cyanglaz** [comment]
This issue seems to come back on iOS 15 +.

Tested on iOS 15, 16, and 17.

**Comment by cyanglaz** [comment]
Internal bug b/292047405 maybe related.

**Comment by cyanglaz** [comment]
@neiljaywarner The regression has been fixed in https://github.com/flutter/engine/pull/44014.
This issue is open to keep track of the issue on iOS 13 and lower.

**Comment by roeyHonig** [comment]
@cyanglaz This is reproducible in iOS16 Flutter 3.1.0 using the example app in webview_flutter plugin.
We can only swipe gesture while on voiceover. 
Direct tapping losses focus entirely.

**Comment by ewelinabd** [comment]
Hi, any progress on this issue

**Comment by PaulinaLapinska** [comment]
@cyanglaz Hi, bump, what is the status with WKWebView?

**Comment by krzysztof-swi** [comment]
The issue can also be reproduced using flutter 3.24.3 and webview_flutter 4.10.
